src/SFA.DAS.LevyTransferMatching.Web/Orchestrators/PledgeOrchestrator.cs

```diff
@@ -38,46 +52,77 @@ public async Task<CreateViewModel> GetCreateViewModel(CreateRequest request)
                 Sectors = cacheItem.Sectors,
                 JobRoles = cacheItem.JobRoles,
                 Levels = cacheItem.Levels,
+                LevelOptions = levels,
+                SectorOptions = sectors,
+                JobRoleOptions = jobRoles
             };
         }
 
         public async Task<AmountViewModel> GetAmountViewModel(AmountRequest request)
         {
             var cacheItem = await RetrievePledgeCacheItem(request.CacheKey);
+            var remainingTransferAllowance = await _accountsService.GetRemainingTransferAllowance(request.EncodedAccountId);
         
             return new AmountViewModel
             {
                 EncodedAccountId = request.EncodedAccountId,
                 CacheKey = request.CacheKey,
                 Amount = cacheItem.Amount.ToString(),
+                RemainingTransferAllowance = remainingTransferAllowance.ToString("N0"),
                 IsNamePublic = cacheItem.IsNamePublic
             };
         }
 
         public async Task<SectorViewModel> GetSectorViewModel(SectorRequest request)
         {
             var cacheItem = await RetrievePledgeCacheItem(request.CacheKey);
+            var sectors = await _tagService.GetSectors();
 
             return new SectorViewModel
             {
                 EncodedAccountId = request.EncodedAccountId,
                 CacheKey = request.CacheKey,
-                Sectors = cacheItem.Sectors
+                Sectors = cacheItem.Sectors,
+                SectorOptions = sectors
             };
         }
 
         public async Task<JobRoleViewModel> GetJobRoleViewModel(JobRoleRequest request)
         {
             var cacheItem = await RetrievePledgeCacheItem(request.CacheKey);
+            var jobRoles = await _tagService.GetJobRoles();
 
             return new JobRoleViewModel
             {
                 EncodedAccountId = request.EncodedAccountId,
                 CacheKey = request.CacheKey,
-                JobRoles = cacheItem.JobRoles
+                JobRoles = cacheItem.JobRoles,
+                JobRoleOptions = jobRoles
             };
         }
 
+        public async Task SubmitPledge(CreatePostRequest request)
+        {
+            var cacheItem = await _cacheStorageService.RetrieveFromCache<CreatePledgeCacheItem>(request.CacheKey.ToString());
+
+            if (cacheItem == null)
+            {
+                throw new InvalidOperationException("Unable to submit pledge due to cache expiry");
+            }
+
+            var pledgeDto = new PledgeDto
+            {
+                Amount = cacheItem.Amount.Value,
+                IsNamePublic = cacheItem.IsNamePublic.Value,
```

Comment:
Should these have a null check on them?

Reply:
Not sure they can be null due to validation, but it might be worth trapping and throwing our own exception rather than getting a NRE